tests/unit/extract/test_group_holdings.py

```diff
@@ -0,0 +1,324 @@
+                            settled_units=150,
+                            cost=CurrencyAndAmount(
+                                amount=10000,
+                                currency="AUD"
+                            ),
+                            cost_portfolio_ccy=CurrencyAndAmount(
+                                amount=0,
+                                currency="AUD"
+                            )
+                        )
+                    ]
+                )
+            }
+        ],
+        [
+            "Standard joining with no portfolio separation",
+            {
+                "MyScope : PortfolioA": MockVersionedResourceListOfPortfolioHolding(
+                    portfolio_holdings=[
+                        PortfolioHolding(
+                            instrument_uid="LUID_12345678",
+                            properties={
+                                "MyProperty": ModelProperty(
+                                    key="MyProperty",
+                                    value=PropertyValue(
+                                        label_value="blah"
+                                    )
+                                )
+                            },
+                            holding_type="P",
+                            units=100,
+                            settled_units=100,
+                            cost=CurrencyAndAmount(
+                                amount=10000,
+                                currency="AUD"
+                            ),
+                            cost_portfolio_ccy=CurrencyAndAmount(
+                                amount=0,
+                                currency="AUD"
+                            )
+                        )
+                    ]
+                ),
+                "MyScope : PortfolioB": MockVersionedResourceListOfPortfolioHolding(
+                    portfolio_holdings=[
+                        PortfolioHolding(
+                            instrument_uid="LUID_12345678",
+                            properties={
+                                "MyProperty": ModelProperty(
+                                    key="MyProperty",
+                                    value=PropertyValue(
+                                        label_value="blah"
+                                    )
+                                )
+                            },
+                            holding_type="P",
+                            units=150,
+                            settled_units=150,
+                            cost=CurrencyAndAmount(
+                                amount=10000,
+                                currency="AUD"
+                            ),
+                            cost_portfolio_ccy=CurrencyAndAmount(
+                                amount=0,
+                                currency="AUD"
+                            )
+                        )
+                    ]
+                )
```

Comment:
I think you should include another (different) instrument here to show that (A,A,B) becomes (2xA, B)